src/libraries/Common/src/System/Net/WebSockets/WebSocketValidate.cs

```diff
@@ -37,7 +47,19 @@ internal static partial class WebSocketValidate
         private static readonly SearchValues<char> s_validSubprotocolChars =
             SearchValues.Create("!#$%&'*+-.0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ^_`abcdefghijklmnopqrstuvwxyz|~");
 
-        internal static void ThrowIfInvalidState(WebSocketState currentState, bool isDisposed, WebSocketState[] validStates)
+        internal static void ThrowIfInvalidSendState(WebSocketState currentState, bool isDisposed)
+            => ThrowIfInvalidState(currentState, isDisposed, s_validSendStates);
+
+        internal static void ThrowIfInvalidReceiveState(WebSocketState currentState, bool isDisposed)
+            => ThrowIfInvalidState(currentState, isDisposed, s_validReceiveStates);
+
+        internal static void ThrowIfInvalidCloseOutputState(WebSocketState currentState, bool isDisposed)
+            => ThrowIfInvalidState(currentState, isDisposed, s_validCloseOutputStates);
+
+        internal static void ThrowIfInvalidCloseState(WebSocketState currentState, bool isDisposed)
+            => ThrowIfInvalidState(currentState, isDisposed, s_validCloseStates);
```

Comment:
Why is adding all these methods beneficial?

Reply:
``WebSocketValidate.ThrowIfInvalidState(currentState, isDisposed, WebSocketValidate.ValidCloseStates);`` felt too long winded. I don't feel strongly about these, will revert 👍